pilot/pkg/xds/ads.go

```diff
@@ -242,6 +245,10 @@ func (s *DiscoveryServer) processRequest(req *discovery.DiscoveryRequest, con *C
 	return s.pushXds(con, push, versionInfo(), con.Watched(req.TypeUrl), request)
 }
 
+func isXdsEvent(req *discovery.DiscoveryRequest) bool {
```

Comment:
I would prefer not to have this logic if possible. I think I know why we do - to avoid treating the REQ as an ACK and being ignored? I think we could also set empty responceNonce as an option?